src/VisualStudio/Core/Def/Implementation/TaskList/ExternalErrorDiagnosticUpdateSource.cs

```diff
@@ -342,17 +345,42 @@ public void AddNewErrors(DocumentId documentId, DiagnosticData diagnostic)
             }).CompletesAsyncOperation(asyncToken);
         }
 
+        private InprogressState BuildInprogressState
+        {
+            get
+            {
+                lock (_gate)
+                {
+                    return _stateDoNotAccessDirectly;
+                }
+            }
+        }
+
+        private void ClearInprogressState()
+        {
+            lock (_gate)
+            {
+                _stateDoNotAccessDirectly = null;
+            }
+        }
+
         private InprogressState GetOrCreateInprogressState()
         {
-            if (_state == null)
+            if (_stateDoNotAccessDirectly == null)
```

Comment:
why is it not safe? the issue was state might created multiple time even if it always return right one. this should never make state to be created multiple time in same session.